drivers/UARTSerial.cpp

```diff
@@ -407,6 +419,17 @@ int UARTSerial::enable_output(bool enabled)
     return 0;
 }
 
+void UARTSerial::init() {
+    for (size_t i = 0; i < sizeof _irq / sizeof _irq[0]; i++) {
+        _irq[i] = NULL;
+    }
+
+    serial_init(&_serial, _serial.pin_tx, _serial.pin_rx);
+    serial_baud(&_serial, _baud);
+    serial_irq_handler(&_serial, SerialBase::_irq_handler, (uint32_t)this);
```

Comment:
Feels to me like the above should be a protected method/subroutine in `SerialBase`, as it's what it does in the constructor.

And for symmetry, the `serial_free` should be in `SerialBase`'s destructor, really, and hence maybe a subroutine by the same logic.. If suspending can shut it down, so should destroying. Although that's more of a potential backward compatibility glitch.